Flaky test: replace.test.ts 'should fail safely when old_string is not found' - LLM bypasses replace by rewriting file

## Problem

The integration test `should fail safely when old_string is not found` in `integration-tests/replace.test.ts` is flaky because the LLM sometimes bypasses the expected failure by creatively rewriting the file.

## Test Description

The test:
1. Creates a file `no_match.txt` with content `"hello world"`
2. Asks the LLM: `replace "goodbye" with "farewell" in no_match.txt`
3. Expects the replace to fail (since "goodbye" is not in "hello world")
4. Asserts the file content remains unchanged

## Failure Mode

Sometimes the LLM, instead of attempting the replace and failing gracefully, decides to "help" by:
- Using `write_file` or similar tool to rewrite the entire file content to include "goodbye"
- Then the replace succeeds

Example failure output:
```
AssertionError: File content was modified! 
Original: "hello world"
Current: "This is a test file that contains the word goodbye again."
```

The new content contains "goodbye" - the LLM created content that would make the replace work.

## Root Cause

This is an **LLM behavior test** that assumes the model will follow the prompt literally. However, LLMs can be creative/adversarial and may:
1. Interpret "replace X with Y" as "make sure the file ends up with Y"
2. Use alternative tools (`write_file`) instead of `replace`
3. Modify the file to make the operation possible

## Proposed Solution

Use the existing `excludeTools` setting to restrict which tools the LLM can use during this test:

```typescript
it('should fail safely when old_string is not found', async () => {
  const rig = new TestRig();
  await rig.setup('should fail safely when old_string is not found', {
    settings: {
      // Prevent LLM from using write tools to bypass the replace failure
      excludeTools: ['write_file', 'write_to_file', 'create_file']
    }
  });
  // ... rest of test
});
```

This approach:
- Allows only `read_file` and `replace` tools (what the test actually needs)
- Prevents the LLM from "creatively" rewriting the file
- Makes the test deterministic

## Files Involved

- `integration-tests/replace.test.ts` - The flaky test
- `integration-tests/test-helper.ts` - TestRig already supports `settings.excludeTools`
- `packages/cli/src/config/settingsSchema.ts` - Defines `excludeTools` setting

## Labels

- bug
- flaky-test
- integration-tests

--- a/integration-tests/replace.test.ts
+++ b/integration-tests/replace.test.ts
@@ -94,7 +94,13 @@ describe('replace', () => {
 
   it('should fail safely when old_string is not found', async () => {
     const rig = new TestRig();
-    await rig.setup('should fail safely when old_string is not found');
+    await rig.setup('should fail safely when old_string is not found', {
+      settings: {
+        // Prevent LLM from using write_file to bypass the replace failure
+        // This ensures the test is deterministic by restricting available tools
+        excludeTools: ['write_file'],
+      },
+    });
     const fileName = 'no_match.txt';
     const fileContent = 'hello world';
     rig.createFile(fileName, fileContent);

